sgkit/stats/regenie.py

```diff
@@ -0,0 +1,781 @@
+    regression technique that produces trait estimates for association
+    tests. These estimates are subtracted from trait values and
+    sampling statistics (p-values, standard errors, etc.) are evaluated
+    against the residuals. See the REGENIE preprint [1] for more details.
+
+    [1] - https://www.biorxiv.org/content/10.1101/2020.06.19.162354v2
+
+    Parameters
+    ----------
+    dosage : str
+        Name of genetic dosage variable.
+    covariates : Union[str, Sequence[str]]
+        Names of covariate variables (1D or 2D).
+    traits : Union[str, Sequence[str]]
+        Names of trait variables (1D or 2D).
+    variant_block_size : Optional[Union[int, Tuple[int]]], optional
+        Number of variants in each block.
+        If int, this describes the number of variants in each block
+        but the last which may be smaller.
+        If Tuple[int, ...], this must describe the desired number of
+        variants in each block individually.
+        Defaults to 1000 or num variants // 2, whichever is smaller.
+    sample_block_size : Optional[Union[int, Tuple[int]]], optional
+        Number of samples in each block.
+        If int, this describes the number of samples in each block
+        but the last which may be smaller.
+        If Tuple[int, ...], this must describe the desired number of
+        samples in each block individually.
+        Defaults to 10 sample blocks split roughly across all possible
+        samples or the number of samples, if that number is < 10.
+    alphas : Optional[Sequence[float]], optional
+        List of alpha values to use for regularization, by default None.
+        If not provided, these will be set automatically based on
+        datasize and apriori heritability assumptions.
+    add_intercept : bool
+        Whether or not to add intercept to covariates, by default True.
+    normalize : bool
+        Rescale genotypes, traits, and covariates to have
+        mean 0 and stdev 1, by default False.
+    orthogonalize : bool
+        **Experimental**: Remove covariates through orthogonalization
+        of genotypes and traits, by default False.
+
+    Returns
+    -------
+    Dataset
+        A dataset containing the following variables:
+        - `base_prediction` (blocks, alphas, samples, outcomes): Stage 1
+            predictions from ridge regression reduction .
+        - `meta_prediction` (samples, outcomes): Stage 2 predictions from
+            the best meta estimator trained on the out-of-sample Stage 1
+            predictions.
+        - `loco_prediction` (contigs, samples, outcomes): LOCO predictions
+            resulting from Stage 2 predictions ignoring effects for variant
+            blocks on held out contigs. This will be absent if the
+            data provided does not contain at least 2 contigs.
+
+    Raises
+    ------
+    ValueError
+        If `G`, `X`, and `Y` do not have the same size along
+        the first (samples) dimension.
+    """
```

Comment:
Add a warning saying that binary traits aren't supported (yet).

Reply:
https://github.com/pystatgen/sgkit/pull/66/commits/3ddbd78756380368cfcbada1e247aa42737eb0ee#diff-74c0b482520e25ae56fbd4e900dede59R783